### PR TITLE
Update Liberty Metrics to MP Telemetry 2.0

### PR DESCRIPTION
**Pull request by Channyboy**
datasource.name for connection pool metrics.

Update FATS to repeat test until EE7. This requires updating format of test structure.

Include metric descriptions in the telemetry common bundle as the descriptions have been updated from mp metrics.

- [x] I have considered the risk of behavior change or other zero migration impact (https://github.com/OpenLiberty/open-liberty/wiki/Behavior-Changes).
- [x] If this PR fixes an Issue, the description includes "Fixes #FILLMEIN" or "Resolves #FILLMEIN" (verify `release bug` label if applicable: https://github.com/OpenLiberty/open-liberty/wiki/Open-Liberty-Conventions).
- [x] If this PR resolves an external Known Issue (including APARS), the description includes "Fixes #FILLMEIN" or "Resolves #FILLMEIN".

################################################################################################

#build

### CODE REVIEW

File: dev/io.openliberty.microprofile.telemetry.internal.common/resources/io/openliberty/microprofile/telemetry/internal/common/resources/MPTelemetry.nlsprops [hunk 1]
```diff
@@ -54,3 +54,35 @@ CWMOT5004.factory.used.without.metadata.useraction=Review your code to see why t
...
 CWMOT5005.mptelemetry.unknown.log.source.useraction=Check the configuration information and specify one of the supported sources for MicroProfile Telemetry.
+
+#-----------------------------------------------------------------------------------------------------------------------------
+# Monitor Metrics messages
+#-----------------------------------------------------------------------------------------------------------------------------
+
+threadpool.activeThreads.description=The number of threads that are actively running tasks.
+threadpool.size.description=The size of the thread pool.
+
+session.activeSessions.description=The number of concurrently active sessions. A session is considered active if the application server is processing a request that uses that user session.
```
Comment (dmuelle) [inline dev/io.openliberty.microprofile.telemetry.internal.common/resources/io/openliberty/microprofile/telemetry/internal/common/resources/MPTelemetry.nlsprops]:
```suggestion
session.activeSessions.description=The number of concurrently active sessions. A session is active when the runtime is processing a request that uses that user session.
```

File: dev/io.openliberty.microprofile.telemetry.internal.common/resources/io/openliberty/microprofile/telemetry/internal/common/resources/MPTelemetry.nlsprops [hunk 2]
```diff
@@ -54,3 +54,35 @@ CWMOT5004.factory.used.without.metadata.useraction=Review your code to see why t
...
+session.invalidated.description=The number of sessions logged out since this metric was enabled.
+session.invalidatedbyTimeout.description=The number of sessions logged out because of a timeout since this metric was enabled.
+session.live.description=The number of users that are currently logged in.
+
+connectionpool.handle.count.description=The number of connections that are in use. This number might include multiple connections that are shared from a single managed connection.
+connectionpool.connection.created.description=The total number of managed connections that were created since the pool creation.
+connectionpool.connection.destroyed.description=The total number of managed connections that were destroyed since the pool creation.
+connectionpool.connection.free.description=The number of managed connections that are available.
+connectionpool.connection.waitTime.description=The amount of time that connection requests waited for a connection.
+connectionpool.connection.useTime.description=The amount of time connections were used for.
```
Comment (dmuelle) [inline dev/io.openliberty.microprofile.telemetry.internal.common/resources/io/openliberty/microprofile/telemetry/internal/common/resources/MPTelemetry.nlsprops]: is this a sum total of all connections or the time for a single connection?

File: dev/io.openliberty.microprofile.telemetry.internal.common/resources/io/openliberty/microprofile/telemetry/internal/common/resources/MPTelemetry.nlsprops [hunk 3]
```diff
@@ -54,3 +54,35 @@ CWMOT5004.factory.used.without.metadata.useraction=Review your code to see why t
...
+connectionpool.connection.count.description=The current sum of managed connections in the pool. This includes managed connections that are available as well as those that are in use. A single managed connection that is shared by multiple connections only counts once.
```
Comment (dmuelle) [inline dev/io.openliberty.microprofile.telemetry.internal.common/resources/io/openliberty/microprofile/telemetry/internal/common/resources/MPTelemetry.nlsprops]:
```suggestion
connectionpool.connection.count.description=The current number of managed connections in the pool. This value includes managed connections that are available and those that are in use. A single managed connection that is shared by multiple connections counts only once.
```

**Review by dmuelle — CHANGES_REQUESTED** [review]
ID review

File: dev/io.openliberty.microprofile.telemetry.internal.common/resources/io/openliberty/microprofile/telemetry/internal/common/resources/MPTelemetry.nlsprops (hunk 2 above)
Comment (Channyboy) [reply dev/io.openliberty.microprofile.telemetry.internal.common/resources/io/openliberty/microprofile/telemetry/internal/common/resources/MPTelemetry.nlsprops]: For a specific connection. (Each metric corresponds to a datasource)

**Review by fmhwong — DISMISSED** [review]
LGTM

File: dev/io.openliberty.microprofile.telemetry.internal.common/resources/io/openliberty/microprofile/telemetry/internal/common/resources/MPTelemetry.nlsprops [hunk 4]
```diff
@@ -54,3 +54,35 @@ CWMOT5004.factory.used.without.metadata.useraction=Review your code to see why t
...
+session.invalidated.description=The number of sessions logged out since this metric was enabled.
+session.invalidatedbyTimeout.description=The number of sessions logged out because of a timeout since this metric was enabled.
+session.live.description=The number of users that are currently logged in.
+
+connectionpool.handle.count.description=The number of connections that are in use. This number might include multiple connections that are shared from a single managed connection.
+connectionpool.connection.created.description=The total number of managed connections that were created since the pool creation.
+connectionpool.connection.destroyed.description=The total number of managed connections that were destroyed since the pool creation.
+connectionpool.connection.free.description=The number of managed connections that are available.
+connectionpool.connection.waitTime.description=The amount of time that connection requests waited for a connection.
+connectionpool.connection.useTime.description=The amount of time connections were used for.
```
Comment (dmuelle) [inline dev/io.openliberty.microprofile.telemetry.internal.common/resources/io/openliberty/microprofile/telemetry/internal/common/resources/MPTelemetry.nlsprops:76]:
```suggestion
connectionpool.connection.useTime.description=The amount of time a connection was used for.
```

File: dev/io.openliberty.microprofile.telemetry.internal.common/resources/io/openliberty/microprofile/telemetry/internal/common/resources/MPTelemetry.nlsprops (hunk 2 above)
Comment (dmuelle) [reply dev/io.openliberty.microprofile.telemetry.internal.common/resources/io/openliberty/microprofile/telemetry/internal/common/resources/MPTelemetry.nlsprops]: added a suggestion

File: dev/io.openliberty.microprofile.telemetry.internal.common/resources/io/openliberty/microprofile/telemetry/internal/common/resources/MPTelemetry.nlsprops (hunk 4 above)
Comment (Channyboy) [inline dev/io.openliberty.microprofile.telemetry.internal.common/resources/io/openliberty/microprofile/telemetry/internal/common/resources/MPTelemetry.nlsprops]: @dmuelle  sorry, I was wrong. A datasource can have multiple connections. (datasource being a database)

Comment (dmuelle) [reply dev/io.openliberty.microprofile.telemetry.internal.common/resources/io/openliberty/microprofile/telemetry/internal/common/resources/MPTelemetry.nlsprops]: so is this the total amount of time for all connections to the configured datasource?

Comment (Channyboy) [reply dev/io.openliberty.microprofile.telemetry.internal.common/resources/io/openliberty/microprofile/telemetry/internal/common/resources/MPTelemetry.nlsprops]: Yup the time spent by connections made to the data source (The metric has a label/tag/key attribute that indicates which data source it is)

Comment (Channyboy) [reply dev/io.openliberty.microprofile.telemetry.internal.common/resources/io/openliberty/microprofile/telemetry/internal/common/resources/MPTelemetry.nlsprops]: @dmuelle

Comment (dmuelle) [reply dev/io.openliberty.microprofile.telemetry.internal.common/resources/io/openliberty/microprofile/telemetry/internal/common/resources/MPTelemetry.nlsprops]: Is it the total concurrent time between the start of the earliest connection and the end of the latest, or the sum of  each individual connection time, regardless of whether they're concurrent (ie connection 1 duration + connection 2 duration + ....connection n duration)

Comment (Channyboy) [reply dev/io.openliberty.microprofile.telemetry.internal.common/resources/io/openliberty/microprofile/telemetry/internal/common/resources/MPTelemetry.nlsprops]: For the `connectionpool.connection.useTime` it is the sum of each. It records the time of each connection .

Comment (dmuelle) [inline dev/io.openliberty.microprofile.telemetry.internal.common/resources/io/openliberty/microprofile/telemetry/internal/common/resources/MPTelemetry.nlsprops]:
```suggestion
connectionpool.connection.useTime.description=The total amount of time that all connections to the data source were in use.
```

File: dev/io.openliberty.microprofile.telemetry.internal.common/resources/io/openliberty/microprofile/telemetry/internal/common/resources/MPTelemetry.nlsprops [hunk 5]
```diff
@@ -54,3 +54,35 @@ CWMOT5004.factory.used.without.metadata.useraction=Review your code to see why t
...
+session.created.description=The number of sessions logged in since this metric was enabled.
+session.invalidated.description=The number of sessions logged out since this metric was enabled.
+session.invalidatedbyTimeout.description=The number of sessions logged out because of a timeout since this metric was enabled.
+session.live.description=The number of users that are currently logged in.
+
+connectionpool.handle.count.description=The number of connections that are in use. This number might include multiple connections that are shared from a single managed connection.
+connectionpool.connection.created.description=The total number of managed connections that were created since the pool creation.
+connectionpool.connection.destroyed.description=The total number of managed connections that were destroyed since the pool creation.
+connectionpool.connection.free.description=The number of managed connections that are available.
+connectionpool.connection.waitTime.description=The amount of time that connection requests waited for a connection.
```
Comment (dmuelle) [inline dev/io.openliberty.microprofile.telemetry.internal.common/resources/io/openliberty/microprofile/telemetry/internal/common/resources/MPTelemetry.nlsprops]:
```suggestion
connectionpool.connection.waitTime.description=The total amount of time that connection requests waited for a connection.
```

Comment (Channyboy) [reply dev/io.openliberty.microprofile.telemetry.internal.common/resources/io/openliberty/microprofile/telemetry/internal/common/resources/MPTelemetry.nlsprops]: This makes it sound like all connections had to wait at some point. Not all connections need to wait. 
Can we remove the `all` ?

**Review by fmhwong — APPROVED** [review]
Lgtm

**Review by dmuelle — APPROVED** [review]
LGTM, thanks

**Pull request merged**